src/rkopenmdao/runge_kutta_integrator.py

```diff
@@ -816,52 +821,58 @@ def _add_functional_part_to_om_vec(
-        self._run_step_preparation_phase(serialized_state)
+        
+        self._serialized_state = serialized_state
         self._run_step_time_integration_phase()
+        self._update_integration_control_step()
         self._run_step_postprocessing_phase()
         self._run_step_functional_phase()
         self._run_step_write_out_phase()
         if self.comm.rank == 0:
             print(f"\nFinishing step {self.options['integration_control'].step} of compute.\n")
         return self._serialized_state, self._error_controller.delta_time_steps, self._error_controller.local_error_norms
 
-    def _run_step_preparation_phase(self, serialized_state):
-        self._update_integration_control_step()
-        self._serialized_state = serialized_state
-
     def _update_integration_control_step(self):
         delta_t = self.options["integration_control"].delta_t
-        step = self.options["integration_control"].step
-        if step == 0:
-            self.options["integration_control"].step_time = self.options["integration_control"].initial_time
-        else:
-            self.options["integration_control"].step_time = self.options["integration_control"].step_time + delta_t
+        self.options["integration_control"].step_time +=  delta_t
+    
+    def _update_error_estimator(self):
+        print(self.options["integration_control"].step)
+        if self.options["integration_control"].step == self.options["integration_control"].initial_step:
+            self._error_controller.reset()
+            print("ERROR ESTIMATOR HAS BEEN RESET")
 
     def _run_step_time_integration_phase(self):
         butcher_tableau: ButcherTableau = self.options["butcher_tableau"]
-        delta_t = self.options["integration_control"].delta_t
+        next_delta_t = self.options["integration_control"].delta_t_suggestion
         accepted = False
-
+        print("current Step time: ", self.options["integration_control"].step_time)
         while not accepted:
-            for stage in range(butcher_tableau.number_of_stages()):
-                self._stage_computation(stage)
-            if self.options["integration_control"].termination_criterion == 'num_steps':
-                temp_serialized_state, delta_t, accepted\
-                    = self._runge_kutta_scheme.compute_step(delta_t, self._serialized_state, self._stage_cache)
-            elif self.options["integration_control"].termination_criterion == 'time_step':
-                temp_serialized_state, delta_t, accepted \
-                    = self._runge_kutta_scheme.compute_step(delta_t, self._serialized_state,
-                                                            self._stage_cache,
-                                                            self.options["integration_control"].remaining_time)
-        self.options["integration_control"].delta_t = delta_t
+            if self.options["integration_control"].termination_criterion.criterion == 'num_steps':
```

Comment:
It seems to me that the only difference between it `if` and `elif` paths is the setting of `self.options["integration_control"].delta_t` (not counting your debug output). Therefore, that's the only part where the `if`-`elif` is necessary, the remaining code can be deduplicated.